src/abstractdataframe/show.jl

```diff
@@ -195,6 +195,11 @@ function _show(io::IO,
     # string or an `e` precedes it.
     alignment_regex_complex = [r"(?<!^)(?<!e)[+-]"]
 
+    # Make sure that `truncate` does not hide the type and the column name.
+    maximum_columns_width = Int[truncate == 0 ? 0 : max(truncate + 1, l, textwidth(t))
+                                for (l, t) in zip(names_len, types_str)]
+
+    # Apply configurations according to the column's type.
     for i = 1:num_cols
         type_i = nonmissingtype(types[i])
```

Comment:
Note that the first condition will be triggered if `types[i]` is `Missing`, but probably this is not problematic right?

Reply:
Indeed! However, it will not matter because all the lines will have the same size and there will be no dot. So, it will not change anything.

Reply:
Yes - that is why I have merged this, but wanted to leave a note if someone came back to this PR. Thanks!